### DISCUSSION TITLE
How to check whether a ui_element is hidden or not?

### DISCUSSION THREAD

**Discussion by mohitcodess** [MyreMylar/pygame_gui#181] [Q&A]
How do we list all the ui_element manged by a manager? how to check which  elements are  hidden?

i want to achieve something like this
> for sprite in manager.get_sprite_group():
>         if sprite.isHidden():
>               print('Hidden ui element")

**Comment by MyreMylar** [comment]
Quick list for all the visible ones with just their surface/image:

```
for sprite_surface in manager.ui_group.visible:
    print('Visible UIElement')
```
That's what I use to blit them.


Slower list for the hidden ones:

```
hidden_sprites = [spr for spr in manager.ui_group.sprites() if not spr.visible]
for sprite in hidden_sprites:
    print('Hidden UIElement with id:', sprite.most_specific_combined_id)
```

That should give you the actual sprite/UIElement. Not tested those myself, just made from looking at the code.

**Reply by MyreMylar** [reply]
I migrated this over to a discussion, mostly because I am testing this new feature of GitHub right now and this seemed an appropriate issue to move.

**Reply by mohitcodess** [reply]
is there a way to find all the data members of a object.. Its not mentioned in the documentation i Guess?
